### DISCUSSION TITLE
Best practice for fitting complex RxODE models

### DISCUSSION THREAD

**Discussion by zebulon2** [nlmixrdevelopment/RxODE#474] [General]
Hi,
I am using RxODE to program complex Systems Biology ODE models, which are much larger in some cases than PK models due to the more mechanistic description of processes. They can contain several dozens equations and parameters. I wanted to have your opinion when it comes to fit some parameters subsets in those models. To do so I usually use an objective function to minimise the error between simulation and observed data for one or several dependent variables, and use an optimisation algorithm from optimx or nloptr packages to perform the parameter optimisation.
Are there some recommendations on best practices? I see that nlme_ode can be used to fit an RxODE model, but I am unsure about the syntax for such kind of models. However the output of the fit function, as well as the performance of nlmixr make it interesting.
Any experience to share in that matter? Thanks a lot for your suggestions.

**Comment by mattfidler** [comment]
Hi @zebulon2,

I have found `nlme` models to not be as performant as other methods.  The `RxODE` model and `nlmixr` models are very similar and there shouldn't be a large transition between the two.  

If you are going to use a systems biology approach, you often do not have between subject variabilities.  If you use nlmixr, use the "focei" method which downgrades to a simple population fit.

In theory, you can also use any outer optimization method that you wish, so if you are not happy with an algorithm or fit you can try another algorithm.  There are pre-coded algorithms like "bobyqa" or others.  

In the future we may be adding forward for population analyses sensitivities to allow a more accurate gradient during optimization.

The new package `rxode2` allows `nlmixr2` syntax models to be used in `rxode2` directly.  We are in the process of submitting the `nlmixr2` package right now and hope to have it on CRAN soon.

**Reply by mattfidler** [reply]
A more complex PBPK example is here:

https://nlmixr2.github.io/nlmixr2/articles/mavoglurant.html

**Reply by zebulon2** [reply]
Thank you very much for the pointers. As you mentioned, virtual populations are not the first priority with such models, but they may come later when doing some systems pharmacology analyses.
So I understand you recommend converting the RxODE model to nlmixr to take advantage of the fitting function with nlmixr, even for non-PK models. For outer optimization with RxODE, I used a log transform for the parameters I want to estimate, as shown in your PBPK example. BOBYQA is one of the algorithms that perform well indeed, but it is easy to automate estimation using a family or diverse algorithms, with various degree of success (as expected with large, non-linear models).

**Reply by mattfidler** [reply]
You could use either approach, of course, it is simply that `nlmixr2` already codes some of them 😄

**Reply by zebulon2** [reply]
Dear Matt, many thanks for the pointers. Is there a documentation available for nlmixr2?

**Reply by mattfidler** [reply]
Indeed, at [nlmixr2.org](https://nlmixr2.org/)
